Access changes for chaplain g.tech and engineers

katnep opened the pull request:
changes made
chaplain has no theatre access anymore
Engineers now have atmos access
g.techs have been stripped of science access

katnep committed:
access tweaks

removes maint access from roles that don't need them, I.E janitor, and
assistant

katnep committed:
assistant tweak

removed checking for assistants have maint config

TalkingCactus commented on code/modules/jobs/job_types/assistant.dm:
```diff
@@ -16,11 +16,7 @@ Assistant
-		. |= list(access_maint_tunnels)
-	else
-		return ..()
+	return ..()
```
This is not something you should be changing. Talk with the host if you want to change something in the configuration files.

Roland410 replied:
While you are somewhat right, I don't think it should ever be enabled, but I might be biased. As I said in another comment, maintenance should be reserved for personnel who HAVE to do something there.

TalkingCactus commented on code/modules/jobs/job_types/cargo_service.dm:
```diff
@@ -267,8 +267,8 @@ Janitor
-	access = list(access_janitor, access_maint_tunnels)
-	minimal_access = list(access_janitor, access_maint_tunnels)
+	access = list(access_janitor)
+	minimal_access = list(access_janitor)
```
Janitors have maintenance access so they can clean up the oil, broken glass, and dried blood; as well as setting mouse traps.

Roland410 replied:
Mouse traps kind of make sense, but the point of maintenance in real life is the authorized personnel use it to maintain wiring/machinery etc.

TalkingCactus reviewed (changes requested):
These changes are either unnecessary or incorrectly executed.

katnep committed:
Revert "assistant tweak"

This reverts commit 63fe914117a9e9c6b52a9effe29c2131c8da7a4e.

katnep committed:
Revert "access tweaks"

This reverts commit 68144943ee0e8b33cd8560a9dd4620532891424c.

katnep committed:
access changes (actually valid edition)

adds atmos access to engineers even on high pop rounds, because atmos
have all engi access and it makes sense to have it the other way round

strips the science access of geneticist because they don't actually need
to be in there even if they are technically researching genetics

katnep committed:
chaplain

removes chaplain theatre access becuase they don't really need it

Roland410 reviewed (commented):
Properly did review below. I sometimes hate Github.

Roland410 commented on code/modules/jobs/job_types/engineering.dm:
```diff
@@ -77,7 +77,7 @@ Station Engineer
 	access = list(access_engine, access_engine_equip, access_tech_storage, access_maint_tunnels,
```
Add atmos access here too.

TalkingCactus replied:
it's already there

Roland410 replied:
My bad, it is indeed there.

katnep replied:
can i just say, some of these access changes seem to be a bit drastic, cook is meant to have morgue access to mince bodies, which if we are trying to justify this with how a space station should work, is what would happen since recycling bodies into food, sci would need tech access since they can use some of the boards in there and no one really uses it as is.

Roland410 replied:
In my opinion cook should not be using those bodies to make food, there are a lot of alternatives. Also scientist doesn't need tech access because that should mainly be for engineers who need to fix a bombed place etc.

katnep replied:
we are on a server with vore, if someone can eat another person, they can grind up souless bodies and eat them

engineers repair the structures, scientists maintain the machinery

Roland410 commented on code/modules/jobs/job_types/medical.dm:
```diff
@@ -127,8 +127,8 @@ Geneticist
 
-	access = list(access_medical, access_morgue, access_chemistry, access_genetics, access_cloning, access_research, access_xenobiology, access_robotics, access_mineral_storeroom, access_tech_storage)
-	minimal_access = list(access_medical, access_morgue, access_genetics, access_cloning, access_research)
+	access = list(access_medical, access_morgue, access_chemistry, access_genetics, access_cloning, access_research, access_mineral_storeroom)
```
No need for ORM access.

TalkingCactus replied:
ORM is for chemistry.

Roland410 reviewed (changes requested):
I think some stuff managed to get out of the PR somehow? Maybe it's just me, but I think we are missing a few commits/changes.
Miners still don't need maint access, remove bartender and cook having morgue access, remove scientist tech storage access and reserve that for RD/heads, same with robo, add security access to detective on skeleton.

katnep committed:
detective sec access

gives detective security access on skeleton crew

Roland410 reviewed (approved):
After finally managing to agree on these, and if Cactus/Pooj say the code is fine, this should be ready to merge.

TalkingCactus commented on code/modules/jobs/job_types/security.dm:
```diff
@@ -124,7 +124,7 @@ Detective
 	outfit = /datum/outfit/job/detective
 
-	access = list(access_sec_doors, access_forensics_lockers, access_morgue, access_maint_tunnels, access_court, access_brig, access_weapons)
+	access = list(access_sec_doors, access_security, access_forensics_lockers, access_morgue, access_maint_tunnels, access_court, access_brig, access_weapons)
```
Detective should not have security access. This gives them access to cells and security equipment (which is undoubtedly better and should be reserved for law enforcement), and other things unnecessary for their job.

katnep replied:
i had added the detective skeleton crew access to security for the fact that security players make up a tiny fraction of cit playerbase, let alone regular ones, so when we get secret with only a detective in sec it makes it easier on the crew so they don't get steam rolled by some guy hacking into the armoury five seconds in and killing them all

Roland410 replied:
I myself have requested these additions for the reasons stated by Sota.

TalkingCactus replied:
This isn't the solution we're looking for. Giving the detective security access will make them feel obligated to conduct arrests and do security's job. Furthermore, it turns the detective into nothing but a security officer with extra access to lucrative equipment. Can't let this stay.

Roland410 replied:
It's only on skeleton though, which shouldn't be a problem anyways?

TalkingCactus replied:
Thing is, then there's no incentive to play security over detective. They're two distinctive roles; merging their duties in such a clear cut way is just repeating the same shit tg ended up with in terms of the geneticist's access. I've no problem with the detective filling the gaps during skeleton crew, but their gear is definitely sufficient to do both jobs. A cell and a stun baton are not required to fulfill those duties.

Roland410 replied:
While I'd normally agree, leaving the detective the only option of gunning down someone who does something bad on lowpop or holding them in a chair or something makes it really unprofessional.

katnep replied:
there is plenty of incentive to play security, if we make it clear that detective still has to take orders from sec and has no authority over them, additionally, in roles like medbay if you choose medical doctor on skeleton crew you still get chem access, you aren't told the medivends and the store room are enough, because on low pop, a stick and a non lethal six shooter is not gonna stop the tag team of rampaging changelings

katnep replied:
in essence this encourages powergaming, because they aren't being given the tools they need, so they are gonna find a way to get them that will land them in trouble with server staff

TalkingCactus replied:
I expect the player to know when it's acceptable to use their gun. They're not without their radio and police baton, either. If you think putting someone in a cell for 1 minute is better than taking them into interrogation and/or just giving them a stern warning, then I don't know what to tell you.

TalkingCactus replied:
Take it to the forum and inform the Host(s)/Head Admin(s) if you really think this is the right thing to do.

katnep committed:
Update security.dm
